apps/backend/src/users/users.service.ts

```diff
@@ -1,17 +1,96 @@
-import { Injectable } from '@nestjs/common';
+import {
+  Injectable,
+  BadRequestException,
+  UnauthorizedException,
+} from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
-import { Repository } from 'typeorm';
+import { MongoRepository } from 'typeorm';
+import { Status } from './types';
 
 import { User } from './user.entity';
+import { UpdateUserDTO } from './users.dto';
+import { ObjectId } from 'mongodb';
 
 @Injectable()
 export class UsersService {
   constructor(
     @InjectRepository(User)
-    private usersRepository: Repository<User>,
+    private usersRepository: MongoRepository<User>,
   ) {}
 
   findAll(): Promise<User[]> {
     return this.usersRepository.find();
   }
+
+  async updateUser(
+    UpdateUserDTO: UpdateUserDTO,
+    userId: string,
+  ): Promise<User> {
+    let id: ObjectId;
+    try {
+      id = new ObjectId(userId);
+    } catch (err) {
+      //TODO maybe check for other errors that might be thrown in line 31
+      //right now this assumes that the error is a BSONError when an id is passed in that's not a 24 character hex
+      throw new BadRequestException(
+        'Invalid user ID format. UserID must be a 24 character hex string, 12 byte Uint8Array, or an integer',
+      );
+    }
+
+    const user: User = await this.usersRepository.findOne({
+      where: {
+        _id: { $eq: id },
+      },
+    });
+
+    if (!user) {
+      throw new BadRequestException(`User ${userId} not found.`);
+    }
+
+    const exampleUser: User = {
+      id: new ObjectId('650f00d4f18cd8be2043e297'),
+      status: Status.ADMIN,
+      firstName: 'jimmy',
+      lastName: 'jimmy2',
+      email: 'jimmy.jimmy2@mail.com',
+      profilePicture: null,
+      linkedin: null,
+      github: null,
+      team: null,
+      role: null,
+    };
+
+    if (
+      exampleUser.status === Status.APPLICANT &&
+      userId !== exampleUser.id.toString()
+    ) {
+      throw new BadRequestException(
+        'Invalid update permissions; applicant cannot update another applicant',
+      );
+    }
+
+    if (
+      (exampleUser.status === Status.MEMBER ||
+        exampleUser.status === Status.ALUMNI) &&
+      user.status === Status.APPLICANT
+    ) {
+      throw new BadRequestException(
+        'Invalid update permissions; members and alumni cannot update applicants',
+      );
+    }
+
+    if (
```

Comment:
:question: does this mean that only admins can update other people's info? If so, do the two checks above (on lines 63 and 72) check anything that this check misses (in other words, is it possible for someone to change someone's info such that it triggers the checks on either 63 or 72 but not the check on line 82)?